src/transaction/fee.rs

```diff
@@ -152,7 +152,17 @@ pub fn charge_fee<S: StateReader>(
         block_context.starknet_os_config.gas_price,
         block_context,
     )?;
-    let actual_fee = min(actual_fee, max_fee) * FEE_FACTOR;
+
+    let actual_fee = if tx_execution_context.version != 0.into() {
```

Comment:
I think this version check is not quite right -- it doesn't allow for version QUERY_VERSION_BASE.